src/main/java/org/gradle/crypto/checksum/Checksum.java

```diff
@@ -52,12 +62,26 @@
         }
     }
 
-    public Checksum() {
-        outputDir = new File(getProject().getBuildDir(), "checksums");
-        algorithm = Algorithm.SHA256;
+    @Inject
+    public Checksum(ObjectFactory objectFactory) {
+        this.objectFactory = objectFactory;
+        this.outputDir = objectFactory.directoryProperty();
+        //TODO use convention when stopping Gradle 5.0 reports
+        outputDir.set(getProject().getLayout().getBuildDirectory().dir("checksums").get());
```

Comment:
:x: don't `.get()` when it is not necessary
```suggestion
        outputDir.set(getProject().getLayout().getBuildDirectory().dir("checksums"));
```